graph/src/data/subgraph/mod.rs

```diff
@@ -666,6 +676,68 @@ impl UnresolvedMapping {
     }
 }
 
+#[derive(Clone, PartialEq, Debug)]
+pub struct UnifiedMappingApiVersion(Option<Version>);
+
+impl UnifiedMappingApiVersion {
+    pub fn equal_or_greater_than(&self, other_version: &'static Version) -> bool {
+        assert!(
+            other_version >= &API_VERSION_0_0_5,
+            "api versions before 0.0.5 should not be used for comparison"
+        );
+        match &self.0 {
+            Some(version) => version >= other_version,
+            None => false,
+        }
+    }
+
+    pub fn try_from_versions<'a>(
+        versions: impl Iterator<Item = &'a Version>,
+    ) -> Result<Self, HashSet<&'a Version>> {
+        let unique_versions: HashSet<&Version> = versions.into_iter().collect();
+
+        let all_below_referential_version = unique_versions.iter().all(|v| *v < &API_VERSION_0_0_5);
+        let all_the_same = unique_versions.len() == 1;
+
+        let unified_version: Option<Version> = match (all_below_referential_version, all_the_same) {
+            (false, false) => return Err(unique_versions),
+            (false, true) => Some(unique_versions.iter().nth(0).unwrap().deref().clone()),
+            (true, true) => None,
+            (true, false) => None,
```

Comment:
This match is nice, lets just collapse the last two cases as `(true, _) => None`.